src/wire/dhcpv4.rs

```diff
@@ -21,6 +22,107 @@ enum_with_unknown! {
     }
 }
 
+/// The possible system architecture types
+#[repr(u16)]
+#[derive(Debug, PartialEq, Eq, Clone, Copy)]
+#[cfg_attr(feature = "defmt", derive(defmt::Format))]
+#[allow(unused)]
+pub enum ClientArchType {
+    /// Intel x86PC
+    X86PC = 0,
+    /// NEC/PC98
+    PC98 = 1,
+    /// EFI Itanium
+    EfiItanium = 2,
+    /// DEC Alpha
+    DecAlpha = 3,
+    /// Arc x86
+    ArcX86 = 4,
+    /// Intel Lean Client
+    IntelLeanClient = 5,
+    /// EFI IA32
+    EfiIA32 = 6,
+    /// EFI BC
+    EfiBc = 7,
+    /// EFI Xscale
+    EfiXscale = 8,
+    /// EFI x86-64
+    EfiX86_64 = 9,
```

Comment:
This is wrong. 7 is EFI x86-64 while 9 is EFI BC, see errata https://www.rfc-editor.org/errata_search.php?rfc=4578.

According to the errata this field should follow the IANA assignments for DHCPv6 options (https://www.iana.org/assignments/dhcpv6-parameters/dhcpv6-parameters.xhtml#processor-architecture), which includes a few additional types (boot from http, other architecture like ARM, RISC-V etc), and deprecates a few ones.

Reply:
Ohh thanks interesting! I never thought of looking for an errata, yeah I will fix that tomorrow~

Reply:
Fixed